### PR TITLE
Improve error message if GFF file is missing attribute col

### PR DESCRIPTION
**Pull request by nsoranzo**
Alternative to #7093.

### CODE REVIEW

File: lib/galaxy/datatypes/util/gff_util.py [hunk 1]
```diff
@@ -42,6 +42,8 @@ def __init__(self, reader, fields, chrom_col=0, feature_col=2, start_col=3, end_
         self.score = self.fields[self.score_col]
 
         # GFF attributes.
+        if 8 >= self.nfields:
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/util/gff_util.py]:
```suggestion
        if self.nfields < 9:
```

File: lib/galaxy/datatypes/util/gff_util.py [hunk 2]
```diff
@@ -42,6 +42,8 @@ def __init__(self, reader, fields, chrom_col=0, feature_col=2, start_col=3, end_
...
+            raise MissingFieldError("No field for attribute column (8)")
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/util/gff_util.py:46]:
```suggestion
            raise MissingFieldError("No field for attribute column (9)")
```
It's the ninth column we're validating, right ?

File: lib/galaxy/datatypes/util/gff_util.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy/datatypes/util/gff_util.py]: I was using the same coding style of the code above, but we could probably update all conditions as you suggested.

File: lib/galaxy/datatypes/util/gff_util.py (hunk 2 above)
Comment (nsoranzo) [reply lib/galaxy/datatypes/util/gff_util.py:46]: Also in this case I'm using the 0-indexed column number as in the code above and as used also in bx-python. Not sure it's useful to fix only in this method.

Comment (mvdbeek) [reply lib/galaxy/datatypes/util/gff_util.py:46]: I think if it's a user-facing exception it'd be clearer to use 1-based counting, but either way is fine.

**Review by mvdbeek — APPROVED** [review]

**Pull request merged**